Dockerfile

```diff
@@ -1,19 +1,91 @@
-FROM node:14-alpine
+# ==
+# builder
+# ==
+
+# application builder
+FROM node:16 AS builder
+
+WORKDIR /usr/app
+
+# copy package.json and package-lock.json files
+COPY package*.json .
+
+# install dependencies including local development tools
+RUN npm install
+
+# copy the rest of the content
+COPY . /usr/app
+
+# disable telemetry when building the app
+ENV NUXT_TELEMETRY_DISABLED=1
+
+# build the application and generate a distribution package
+RUN npm run build
+
+# ==
+# development
+# ==
+# application for development purposes
+FROM node:16 AS dev
 
-# directory for the app in the container
 WORKDIR /usr/app
 
-# copies all the app's files from host into the container folder which
-# might include the node_modules dir if npm install executed in the host
+ENV NODE_ENV=development
+ENV CYPRESS_INSTALL_BINARY=0
+
+# copy files from local machine
 COPY . /usr/app
 
-# removes any existing node_modules folder
-# this prevents the host's node_modules from being used in the container
-# which could cause issues with native binaries such as node_sass.
-RUN rm -rf /usr/app/node_modules/
+# disable telemetry when building the app
+ENV NUXT_TELEMETRY_DISABLED=1
+
+# install dependencies (development dependencies included)
+RUN npm install
+
+# expose port 8433
+EXPOSE 8433
 
+# run the application in development mode
+ENTRYPOINT [ "npm", "run", "dev" ]
+
+# ==
+# production
+# ==
+# application package (for production)
+FROM node:16 AS app
+
+WORKDIR /usr/app
+
+ENV NODE_ENV=production
 ENV CYPRESS_INSTALL_BINARY=0
```

Comment:
We have replaced cypress with Playwright for e2e tests, so we don't need the `CYPRESS_INSTALL_BINARY` variable anymore.

Reply:
Do we need the corresponding playwright variable?

Reply:
Playwright has `PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD`: https://playwright.dev/docs/browsers/#skip-browser-downloads

Reply:
@obulat @zackkrida env variable replaced